### ISSUE TITLE
parent structs error when `#:type-name` is provided

### ISSUE THREAD

**Issue by florence** [racket/typed-racket#506]
### What version of Racket are you using?
6.0.8.2 (Current TR head as of yesterday)

### What program did you run?
```
#lang typed/racket
(struct a ([f : Integer])
  #:type-name A)
(struct b a ())
```
### What should have happened?
The program should have behaved like this one:

```
#lang typed/racket
(struct a ([f : Integer]))
(struct b a ())
```
### If you got an error message, please include it here.

```
Type Checker: parent type not a valid structure name: a in: a
```

**Comment by default-kramer** [comment]
I think I found a fix for this. [This line](https://github.com/racket/typed-racket/blob/c523887c82f0977940426bcf48dd09d8f6389838/typed-racket-lib/typed-racket/typecheck/tc-structs.rkt#L137) calls `register-resolved-type-alias` with the result of `struct-names-type-name`. I duplicated that call except with the result of `struct-names-struct-names` and the OP's test case passes.

I ran `raco test ./` in C:\Program Files\Racket\share\pkgs\typed-racket-lib and all the tests seemed to pass.

If this fix looks good, I would be willing to make a PR. But it might be easier if a regular contributor does it.
